sdks/python/apache_beam/coders/row_coder.py

```diff
@@ -181,9 +192,11 @@ def encode_to_stream(self, value, out, nested):
         for i, is_null in enumerate(nulls):
           words[i // 8] |= is_null << (i % 8)
 
+    attrs = self._enc_pos_idx(attrs)
+    fields = self._enc_pos_idx(self.schema.fields)
```

Comment:
We need to avoid using logic like this to determine the field order in the encode_to_stream method. This implementation is essentially determining the field order dynamically for every encoded element. It should be possible to use something like `np.argsort` to determine the order one time (likely in the `__init__` method), and then reference that order in the encode_to_stream method.

Please note we'll also need to reference that order in the decode_from_stream method. RowCoder should respect the encoding positions when encoding _and_ decoding.

Reply:
I use `np.argsort` instead, and also implemented on both encoding and decoding.